views/js/pciCreator/ims/mathEntryInteraction/creator/widget/states/Map.js

```diff
@@ -73,48 +78,136 @@ define([
             }
+        } else {
+            newCorrectAnswer = '';
+            this.toggleResponseMode(false);
+        }
+
+        this.correctResponses.push(newCorrectAnswer);
+    }
 
-            self.correctResponses.push(newCorrectAnswer);
-            self.renderForm(self.correctResponses);
+    MathEntryInteractionStateResponse.prototype.createScoreResponse = function createScoreResponse() {
+        var $container = this.widget.$container;
+        var $input = $container.find('.math-entry-input');
+
+        if ($container.find('.math-entry-response-wrap').length > 0) {
+            return false
+        }
+
+        var $interaction = this.widget.element;
+        var response = $interaction.getResponseDeclaration();
+        this.$responseCorrectTitle = $('<span>', {
+            'class': 'math-entry-response-title math-entry-response-correct'
+        }).html('Correct');
+        $input[0].parentNode.prepend(this.$responseCorrectTitle[0]);
+        $input[0].parentNode.className = 'math-entry-correct-wrap';
+        $input[0].parentNode.dataset.index = 0;
+        this.$scoreDiv = $('<div>', {
+            'class': 'math-entry-score-wrap math-entry-response-correct'
         });
+        $input[0].parentNode.parentNode.insertBefore(this.$scoreDiv[0], $input.nextSibling);
+        this.$scoreDiv.after($addAlternativeBtn);
+        $('.math-entry-correct-wrap, .math-entry-score-wrap').wrapAll('<div class="math-entry-response-wrap"></div>');
+        var pairId = $interaction.attr('responseIdentifier');
+    
+        this.$responseCorrectScore = $('<span>', {
+            'class': 'math-entry-score-title math-entry-response-correct'
+        }).html('Score');
+        this.$scoreDiv.append(scoreTpl({
+            serial: response.serial,
+            mathIdentifier: pairId,
+            placeholder: response.getMappingAttribute('defaultValue')
+        }));
+        this.$scoreDiv.prepend(this.$responseCorrectScore);
+
+        //add placeholder text to show the default value
+        var $scores = $container.find('.math-entry-response-wrap .math-entry-score-input');
+        $scores.on('click', function (e) {
+            e.stopPropagation();
+            e.preventDefault();
+        });
+
+        this.widget.on('mappingAttributeChange', function (data) {
+            if (data.key === 'defaultValue') {
+                $scores.attr('placeholder', data.value);
+            }
+        });
+
+        //init form javascript
+        formElement.initWidget($container);
+        formElement.setChangeCallbacks($container, response,
+            _.assign({
+                mathEntryScoreInput: function (response, value) {
+                    var key = $(this.widget).data('for');
+                    if (value === '') {
+                        response.removeMapEntry(key);
+                    } else {
+                        response.setMapEntry(key, value, true);
+                    }
+
+                }
+            })
```

Comment:
**question:** What is the benefit of using `_.assign()` without a target?

**suggestion:** It could be simplified using an object literal, ES6+ shorthand notation can be applied too
```suggestion
            {
                mathEntryScoreInput(response, value) {
                    const key = $(this.widget).data('for');
                    if (value === '') {
                        response.removeMapEntry(key);
                    } else {
                        response.setMapEntry(key, value, true);
                    }
                }
            }
```